### ISSUE TITLE
Add a sink.set_speed function that keeps original perceived pitch

### ISSUE THREAD

**Issue by sloganking** [RustAudio/rodio#542]
[sink.set_speed](https://docs.rs/rodio/0.17.3/rodio/struct.Sink.html#method.set_speed) perceptibly makes the audio higher pitch when the speed is above `1.0` and perceptually lower pitch when lower when the speed is below `1.0`. Most modern video players and other user facing software that play sound, adjust the pitch when audio is sped up or slowed down so that it is perceptibly the same pitch to human ears. I suggest adding a function that does this. Allowing for audio to be sped up without sounding different.

**Comment by tramhao** [comment]
I suggest use a simple algorithm like scaletempo. Or could be rubberband.

**Comment by dvdsk** [comment]
I agree that this would be a good addition. Unfortunately Rodio is a bit low on maintainers.

If anyone feels like giving it a go I would be more then willing to review/mentor a PR and eventually merge it.
